component/src/main/java/io/siddhi/extension/io/http/source/HttpSSESource.java

```diff
@@ -0,0 +1,427 @@
+        this.workerThread = Integer.parseInt(optionHolder
+                .validateAndGetStaticValue(HttpConstants.WORKER_COUNT, DEFAULT_WORKER_COUNT));
+        this.httpMethod = optionHolder.validateAndGetStaticValue(HttpConstants.METHOD, HttpConstants.HTTP_METHOD_GET);
+        this.clientStoreFile = optionHolder.validateAndGetStaticValue(HttpConstants.CLIENT_TRUSTSTORE_PATH_PARAM,
+                HttpSinkUtil.trustStorePath(configReader));
+        this.clientStorePass = optionHolder.validateAndGetStaticValue(HttpConstants.CLIENT_TRUSTSTORE_PASSWORD_PARAM,
+                HttpSinkUtil.trustStorePassword(configReader));
+        this.clientBootstrapConfiguration = optionHolder
+                .validateAndGetStaticValue(HttpConstants.CLIENT_BOOTSTRAP_CONFIGURATION, EMPTY_STRING);
+        this.userName = optionHolder.validateAndGetStaticValue(HttpConstants.RECEIVER_USERNAME, EMPTY_STRING);
+        this.userPassword = optionHolder.validateAndGetStaticValue(HttpConstants.RECEIVER_PASSWORD, EMPTY_STRING);
+
+        authType = validateAndGetAuthType();
+
+        String scheme = configReader.readConfig(HttpConstants.DEFAULT_SOURCE_SCHEME, HttpConstants
+                .DEFAULT_SOURCE_SCHEME_VALUE);
+        int port;
+        boolean isSecured;
+        if (HttpConstants.SCHEME_HTTPS.equals(scheme)) {
+            port = Integer.parseInt(configReader.readConfig(HttpConstants.HTTPS_PORT, HttpConstants.HTTPS_PORT_VALUE));
+            isSecured = true;
+        } else {
+            port = Integer.parseInt(configReader.readConfig(HttpConstants.HTTP_PORT, HttpConstants.HTTP_PORT_VALUE));
+            isSecured = false;
+        }
+
+        this.serviceDeploymentInfo = new ServiceDeploymentInfo(port, isSecured);
+        this.listenerUrl = optionHolder.validateAndGetOption(HttpConstants.LISTENER_URL).getValue();
+        this.httpConnectorRegistry = HttpSSESourceConnectorRegistry.getInstance();
+        this.requestedTransportPropertyNames = requestedTransportPropertyNames.clone();
+        this.httpConnectorFactory = createConnectorFactory(configReader);
+        this.connectionPoolConfiguration = createPoolConfigurations(optionHolder);
+        this.clientConnector = createClientConnector();
+        initMetrics();
+        return null;
+    }
+
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[0];
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback, State state) throws ConnectionUnavailableException {
+        this.httpSSEResponseConnectorListener =
+                new HttpSSEResponseConnectorListener(workerThread, sourceEventListener, streamID,
+                        requestedTransportPropertyNames, siddhiAppName);
+        this.httpConnectorRegistry.registerSourceListener(httpSSEResponseConnectorListener, streamID);
+        HTTPSourceRegistry.registerSSESource(streamID, this);
+
+        String contentType = HttpConstants.APPLICATION_JSON;
```

Comment:
@cazci shall we make content-type parameterized? since in sse, theres are several types including 'text/event-stream'